aidants_connect_habilitation/forms.py

```diff
@@ -235,11 +230,45 @@ class PersonnelForm:
     DPO_FORM_PREFIX = "dpo"
     AIDANTS_FORMSET_PREFIX = "aidants"
 
+    @property
+    def non_field_errors(self):
+        errors = self.manager_form.non_field_errors().copy()
+
+        for error in self.data_privacy_officer_form.non_field_errors():
+            errors.append(error)
+
+        for error in self.aidants_formset.non_form_errors():
+            errors.append(error)
+
+        return errors
+
     def __init__(self, **kwargs):
         def merge_kwargs(prefix):
             previous_prefix = kwargs.get("prefix")
+            local_kwargs = {}
+
+            form_kwargs_prefixes = {
+                self.MANAGER_FORM_PREFIX,
+                self.DPO_FORM_PREFIX,
+                self.AIDANTS_FORMSET_PREFIX,
+            }
+
+            for k, v in kwargs.items():
+                """
+                Let us dispatch form kwargs to specific forms by using their prefixes.
+
+                For instance, PersonnelForm(dpo_instance=some_instance) will
+                disptach to DataPrivacyOfficerForm(instance=some_instance).
+                """
+                kwarg_prefix = k.split("_")
+                if len(kwarg_prefix) > 1 and kwarg_prefix[0] in form_kwargs_prefixes:
+                    if kwarg_prefix[0] == prefix:
+                        local_kwargs[k.removeprefix(f"{prefix}_")] = v
```

Comment:
Il y a une erreur ici dans les tests, `k` serait une chaîne et n'aurait pas de méthode `removeprefix`.

Reply:
Ah, c'est une fonction de 3.9, mais la prod en 3.7 et la CI en 3.7 aussi.
(Cette revue de PR est pleine de "ah" a posteriori, desolée pour le spam ^^)

Reply:
Mais si j'ai bien compris, ce que tu essaies de choper là c'est justement ce que j'ai appelé `kwarg_field` dans mon commentaire précédent.

Reply:
Ah ouais, chuis dans le turfu, moi aussi…
Bah ici, le but, c'est de pouvoir passer des paramètres à l'`__init__` d'un des 3 `Form` seulement plutôt que de le passer aux 3. [Par exemple avec `instance`](https://docs.djangoproject.com/en/4.0/topics/forms/modelforms/#the-save-method) qui spécifie l'instance du modèle que le form doit modifier en base. Avec ce traitement, si je passer un arguement à un form spécific. Par exemple, si je passe `dpo_instance=a`, ça passé comme ça `self.data_privacy_officer_form(instance=a)`